lib/backend/kubernetes/kubernetes.go

```diff
@@ -0,0 +1,280 @@
+// local SQLite database. In that case, the agent reads local database and
+// creates a copy of the keys in Kube Secret.
+func (b *Backend) Exists(ctx context.Context) bool {
+	b.mu.Lock()
+	defer b.mu.Unlock()
+
+	_, err := b.getSecret(ctx)
+	return err == nil
+}
+
+// Get reads the secret and extracts the key from it.
+// If the secret does not exist or the key is not found it returns trace.Notfound,
+// otherwise returns the underlying error.
+func (b *Backend) Get(ctx context.Context, key []byte) (*backend.Item, error) {
+	b.mu.Lock()
+	defer b.mu.Unlock()
+
+	return b.readSecretData(ctx, key)
+}
+
+// Create creates item
+func (b *Backend) Create(ctx context.Context, i backend.Item) (*backend.Lease, error) {
+	b.mu.Lock()
+	defer b.mu.Unlock()
+
+	return b.updateSecretContent(ctx, i)
+}
+
+// Put puts value into backend (creates if it does not exist, updates it otherwise)
+func (b *Backend) Put(ctx context.Context, i backend.Item) (*backend.Lease, error) {
+	b.mu.Lock()
+	defer b.mu.Unlock()
+
+	return b.updateSecretContent(ctx, i)
+}
+
+// PutRange receives multiple items and upserts them into the Kubernetes Secret.
+// This function is only used when the Agent's Secret does not exist, but local SQLite database
+// has identity credentials.
+// TODO(tigrato): remove this once the compatibility layer between local storage and
+// Kube secret storage is no longer required!
+func (b *Backend) PutRange(ctx context.Context, items []backend.Item) error {
+	b.mu.Lock()
+	defer b.mu.Unlock()
+
+	_, err := b.updateSecretContent(ctx, items...)
+	return err
+}
+
+// getSecret reads the secret from K8S API.
+func (b *Backend) getSecret(ctx context.Context) (*corev1.Secret, error) {
+	secret, err := b.k8sClientSet.
+		CoreV1().
+		Secrets(b.namespace).
+		Get(ctx, b.secretName, metav1.GetOptions{})
+
+	if err == nil {
+		return secret, nil
+	}
+	var kubeErr *kubeerrors.StatusError
+	if errors.As(err, &kubeErr) && kubeErr.ErrStatus.Code == http.StatusNotFound {
+		return nil, trace.NotFound("secret %v not found", b.secretName)
+	}
+	return nil, trace.Wrap(err)
+}
+
+// readSecretData reads the secret content and extracts the content for key.
+// returns an error if the key does not exist or the data is empty.
+func (b *Backend) readSecretData(ctx context.Context, key []byte) (*backend.Item, error) {
+	secret, err := b.getSecret(ctx)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	data, ok := secret.Data[backendKeyToSecret(key)]
+	if !ok || len(data) == 0 {
+		return nil, trace.NotFound("key [%s] not found in secret %s", string(key), b.secretName)
+	}
+
+	return &backend.Item{
+		Key:   key,
+		Value: data,
+	}, nil
+}
+
+func (b *Backend) updateSecretContent(ctx context.Context, items ...backend.Item) (*backend.Lease, error) {
+	// FIXME(tigrato):
+	// for now, the agent is the owner of the secret so it's safe to replace changes
+
+	secret, err := b.getSecret(ctx)
+	if err != nil && trace.IsNotFound(err) {
```

Comment:
I think we can drop the `err != nil` here
`trace.IsNotFound` will return false if `err == nil`